src/extensions/hive/components/SaveToHiveModal.tsx

```diff
@@ -0,0 +1,394 @@
+
+  dashboard = cachedDashboard ? cachedDashboard : dashboard;
+  const tabCount = 2;
+  const title = dashboard?.title;
+  const existingSolutionId = dashboard?.extensions?.solutionsHive?.uuid;
+  const initialTabIndex = existingSolutionId ? tabCount - 1 : 0;
+
+  const [selectedFile, setSelectedFile] = useState();
+  const [isSelected, setIsSelected] = useState(false);
+  // const [overwriteExistingDashboard, setOverwriteExistingDashboard] = React.useState(false);
+  const [tabIndex, setTabIndex] = useState(initialTabIndex);
+  const [dbConnection, setDbConnection] = useState(connection);
+  const [hasPublished, setHasPublished] = useState(false);
+  const [solutionId, setSolutionId] = useState(0);
+  const [image, setImage] = useState('');
+  const [domain, setDomain] = useState(HiveSolutionDomain.Private);
+  // console.log('tabIndex: ', tabIndex);
+
+  const [hiveSolutionInfoBeenCalled, setHiveSolutionInfoBeenCalled] = useState(false);
+  const [hiveSolutionInfo, setHiveSolutionInfo] = useState({});
+
+  useEffect(() => {
+    const getHiveSolution = async () => {
+      console.log('in getHiveSolution');
+      const uri = config('GALLERY_GRAPHQL_URL');
+      console.log('hive uri is: ', uri);
+      fetch(uri, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: auth.getIdToken() ? `Bearer ${auth.getIdToken()}` : '',
+        },
+        body: JSON.stringify({
+          query: GetSolutionById,
+          variables: { id: existingSolutionId },
+        }),
+      })
+        .then(handleErrors)
+        .then(async (res) => {
+          const jsonResponse = await res.json();
+          const solution = jsonResponse?.data?.solution || {};
+          // console.log('Solution info is: ', solution);
+          setHiveSolutionInfo(solution);
+          if (domain !== solution.domain) {
+            setDomain(solution.domain);
+          }
+          if (image !== solution.image) {
+            setImage(solution.image || '');
+          }
+        })
+        .catch((error) => {
+          console.log(`Error fetching Solution info for Solution ID ${existingSolutionId}`, error);
+        });
+    };
+
+    if (!hiveSolutionInfoBeenCalled) {
+      setHiveSolutionInfoBeenCalled(true);
+      if (existingSolutionId) {
+        console.log('Calling getHiveSolution with solutionId: ', existingSolutionId);
+        try {
+          getHiveSolution();
+        } catch (e) {
+          console.log(`Error fetching Solution info for Solution ID ${existingSolutionId}`, e);
+        }
+      }
+    }
+  }, [hiveSolutionInfo, hiveSolutionInfoBeenCalled, existingSolutionId]);
+
+  const existingDbName = existingSolutionId ? connection.database : null;
+
+  const lastStep = () => tabIndex === tabCount - 1;
+  const firstStep = () => tabIndex === 0;
+  const onPublishStep = () => lastStep() && !hasPublished;
+
+  const { driver } = useContext<Neo4jContextState>(Neo4jContext);
```

Comment:
use connection module here

Reply:
digging deeper, we aren't even using `driver` anymore.  It's passed down but never used.  I'll remove the reference

Reply:
references to `driver` removed